ynr/apps/parties/managers.py

```diff
@@ -0,0 +1,157 @@
+from datetime import date
+import re
+
+from dateutil import parser
+
+from django.utils import timezone
+from django.db import models
+from django.conf import settings
+
+from .constants import JOINT_DESCRIPTION_REGEX
+
+
+class PartyQuerySet(models.QuerySet):
+    def active_for_date(self, date=None):
+        if not date:
+            date = timezone.now()
+        qs = self.filter(date_registered__lte=date)
+        qs = qs.filter(
+            models.Q(date_deregistered__gte=date)
+            | models.Q(date_deregistered=None)
+        )
+        return qs
+
+    def current(self):
+        return self.active_for_date()
+
+    def register(self, register):
+        return self.filter(register=register)
+
+    def order_by_memberships(self, date=None, nocounts=False):
+        qs = self
+        if date:
+            qs = qs.filter(
+                membership__post_election__election__election_date=date
+            )
+            qs = qs.annotate(
+                candidate_count=models.Count("membership")
+            ).order_by("-candidate_count", "name")
+
+        if nocounts:
+            qs = qs.filter(
+                ~models.Q(
+                    membership__post_election__election__election_date=date
+                )
+            ).annotate(candidate_count=models.Value(0, models.IntegerField()))
+
+        return qs
+
+    def party_choices_basic(self):
+        result = list(self.order_by("name").values_list("ec_id", "name"))
+        result.insert(0, ("", ""))
+        return result
+
+    def party_choices(
+        self,
+        include_descriptions=True,
+        exclude_deregistered=False,
+        include_description_ids=False,
+        include_non_current=True,
+    ):
+        # For various reasons, we've found it's best to order the
+        # parties by those that have the most candidates - this means
+        # that the commonest parties to select are at the top of the
+        # drop down.  The logic here tries to build such an ordered
+        # list of candidates if there are enough that such an ordering
+        # makes sense.  Otherwise the fallback is to rank
+        # alphabetically.
+        from popolo.models import Membership
+
+        candidacies_ever_qs = (
+            self.annotate(candidate_count=models.Count("membership")).order_by(
+                "-candidate_count", "name"
+            )
+            # .only("end_date", "name")
```

Comment:
There are 3 places in this file where we've commented out a `.only("end_date", "name")`. Is this intentional? If they're not needed any more, lets actually delete them.